pythonwhat/State.py

```diff
@@ -35,6 +40,7 @@ def __len__(self):
         return len(self._items)
 
 
+@parameters_attr
```

Comment:
I added some reasoning behind this decorator in the docstring of `parameters_attr` in protowhat to make it easy to understand why this was done in the future (when we want to think of a solution that eliminates this trade-off).

Reply:
To have this value available on all State subclasses and their instances without recalculation or decorator State in protowhat could get a metaclass with a parameters property that memoizes the value.
I chose this because it's easier to understand, but one metaclass in protowhat would prevent having to think about it in xwhats.